pkg/kudoctl/cmd/convert/convert.go

```diff
@@ -0,0 +1,196 @@
+package convert
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/yaml"
+
+	kudo "github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+	"github.com/kudobuilder/kudo/pkg/util/helm"
+)
+
+// Options are configuration options for this command
+type Options struct {
+	ChartImportPath string
+	OutputPath      string
+}
+
+// DefaultOptions are options with the defaults applied
+var DefaultOptions = &Options{}
+
+// Run runs the convert command
+func Run(cmd *cobra.Command, args []string, options *Options) error {
+	chart, e := helm.LoadChart(options.ChartImportPath)
+	if e != nil {
+		return e
+	}
+	bundle, e := helm.ToBundle(chart)
+	if e != nil {
+		return e
+	}
+	if b := exists(options.OutputPath); b {
+		return fmt.Errorf("bundle output dir %v shouldn't exist", options.OutputPath)
+	}
+	e = os.Mkdir(options.OutputPath, 0755)
+	if e != nil {
+		return e
+	}
+
+	bundleParams := bundle.Parameters
+	bundle.Parameters = make([]kudo.Parameter, 0)
+	b, e := yaml.Marshal(bundle)
+	if e != nil {
+		return e
+	}
+	e = ioutil.WriteFile(fmt.Sprintf("%v/operator.yaml", options.OutputPath), b, os.ModePerm)
+	if e != nil {
+		return e
+	}
+	type P struct {
+		Default     string `json:"default,omitempty"`
+		Description string `json:"description,omitempty"`
+		Trigger     string `json:"trigger,omitempty"`
+	}
+	mapParams := make(map[string]P)
+	for _, param := range bundleParams {
+		mapParams[param.Name] = P{
+			Default:     param.Default,
+			Description: param.Description,
+			Trigger:     param.Trigger,
+		}
+	}
+	b, e = yaml.Marshal(mapParams)
+	if e != nil {
+		return e
+	}
+	e = ioutil.WriteFile(fmt.Sprintf("%v/params.yaml", options.OutputPath), b, os.ModePerm)
+	if e != nil {
+		return e
+	}
+
+	e = os.Mkdir(options.OutputPath+"/templates", 0755)
+	if e != nil {
+		return e
+	}
+	for _, template := range chart.GetTemplates() {
+		//ignore test files
+		if strings.HasPrefix(template.GetName(), "templates/tests") {
+			continue
+		}
+		e = ioutil.WriteFile(fmt.Sprintf("%v/%v", options.OutputPath, template.GetName()), template.GetData(), os.ModePerm)
+		if e != nil {
+			return e
+		}
+	}
+
+	// e = copyDirectory(fmt.Sprintf("%v/templates", options.ChartImportPath), fmt.Sprintf("%v/templates", options.OutputPath))
+
+	return e
+}
+
+func copyDirectory(scrDir, dest string) error {
```

Comment:
functions specific to other topics (like file ops) should belong in fileops package.

Reply:
which is all functions below in this file